station/submitobs.py

```diff
@@ -88,41 +146,45 @@ def submit_observation(data: SubmitRequestData):
         "Authorization": header_value
     }
 
-    files = {
-        "file": (
-            filename,
-            file_obj,
-            "image/png",
-            {}
-        )
-    }
+    # Check if notes are a valid JSON
+    logging.debug(f"Meta-data = {data.config}")
 
-    logging.info("Submitting observation, url=%s file=%s" % (url, filename))
+    logging.info(f"Submitting observation, url={url}, file(s)={data.image_path}")
 
-    resp = requests.post(url, form_data, headers=headers, files=files)
+    try:
+        resp = requests.post(url, form_data, headers=headers, files=files)
+    except requests.exceptions.ConnectionError:
+        return f"Unable to connect to {url}, connection refused"
     if (resp.status_code >= 400):
         logging.warning("Response status: %d" % resp.status_code)
     else:
         logging.info("Response status: %d" % resp.status_code)
 
     # Logging extra details in case of failed submission.
-    if (resp.status_code != 200):
+    if (resp.status_code != 204):
         # On info, just log the field names in what we sent. On debug, log also the content and the whole response.
         logging.info("Submission details: headers: %s, form: %s" % (",".join(headers.keys()), ",".join(form_data.keys())) )
         logging.debug("headers=%s" % headers)
         logging.debug("form=%s" % form_data)
-        logging.debug("Response details: %s" % repr(resp))
+        logging.debug("Response details: %s" % resp.text)
+        return f"Upload failed, status code {resp.status_code}"
+    else:
+        logging.info("Upload successful.")
+
+    # All seems to be OK
+    return None
 
 if __name__ == '__main__':
     if len(sys.argv) < 4:
```

Comment:
It has grown a lot. We need to refactor it using the `argparse` module. But maybe in a separate issue?

Reply:
There's #57 created for that already.